src/erlang.zig

```diff
@@ -6,37 +6,73 @@ const std = @import("std");
 pub const receiver = @import("receiver.zig");
 pub const sender = @import("sender.zig");
 
-// TODO: move these elsewhere, maybe make them into parameters
-pub const process_name = "server";
-pub const server_name = process_name ++ "@localhost";
-
 pub const Send_Error = sender.Error || error{
     // TODO: rid the world of these terrible names
     new_with_version,
     reg_send_failed,
 };
 
 pub const Node = struct {
+    const name_length = 10;
+    pub const max_buffer_size = 50;
     c_node: ei.ei_cnode,
     fd: i32,
-    node_name: [:0]const u8 = "node",
-    cookie: [:0]const u8 = "cookie",
+    node_name: [name_length:0]u8,
+
+    pub fn init(cookie: [:0]const u8) !Node {
+        var tempNode: Node = .{
+            .c_node = undefined,
+            .fd = undefined,
+            .node_name = .{0} ** name_length,
+        };
+
+        var src_node_name: [name_length / 2]u8 = undefined;
+        std.crypto.random.bytes(&src_node_name);
+        const encoder = std.base64.Base64Encoder.init(std.base64.standard_alphabet_chars, null);
+        _ = encoder.encode(&tempNode.node_name, &src_node_name);
```

Comment:
What is the return of this function? Zig's docs seem to be offline for me at the moment.

Reply:
You can run `zig std` to look at the docs in your own machine.

To answer the question: it returns a slice of the bytes that were actually written, since it may write fewer bytes than the destination slice can hold.